tests/fetchGit.sh

```diff
@@ -123,7 +129,7 @@ path2=$(nix eval --impure --raw --expr "(builtins.fetchGit file://$repo).outPath
 # Using local path with branch other than 'master' should work when clean or dirty
 path3=$(nix eval --impure --raw --expr "(builtins.fetchGit $repo).outPath")
 # (check dirty-tree handling was used)
-[[ $(nix eval --impure --raw --expr "(builtins.fetchGit $repo).rev") = 0000000000000000000000000000000000000000 ]]
+[[ $(nix eval --impure --expr "(builtins.fetchGit $repo).rev or null") = null ]]
```

Comment:
This breaks compatibility since there might be projects that expect that `rev` is always set. `fetchGit` (but not `fetchTree`) on dirty trees should continue to return `000...`.

Reply:
so, partly revert my change to ensure fetchGit returns an empty rev here?

Reply:
Yes please.

Reply:
Fixed. However I'm well-aware that my current fix is fairly ugly since I didn't manage to allocate a new attr to `v` in `prim_fetchGit`. If you have a better solution, please tell me :)